File: frontend/lib/models/round_participant_info.dart

class RoundParticipantInfo {
  final int roundParticipantId;
  final int userId;
  final String picture;
  final String statusTag;
  final double taskProgress;

  RoundParticipantInfo({
    required this.roundParticipantId,
    required this.userId,
    required this.picture,
    required this.statusTag,
    required this.taskProgress,
  });

  factory RoundParticipantInfo.fromJson(Map<String, dynamic> json) {
    return RoundParticipantInfo(
        roundParticipantId: json['roundParticipantId'],
        userId: json['userId'],
        picture: json['picture']??"",
        statusTag: json['statusTag'],
        taskProgress: json['taskProgress'],
    );
  }
}

File: backend/src/main/java/ssu/groupstudy/domain/round/dto/response/RoundParticipantInfo.java
package ssu.groupstudy.domain.round.dto.response;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import ssu.groupstudy.domain.round.entity.RoundParticipantEntity;
import ssu.groupstudy.domain.common.enums.StatusTag;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class RoundParticipantInfo {
    private Long roundParticipantId;
    private Long userId;
    private String picture;
    private StatusTag statusTag;
    private double taskProgress;

    private RoundParticipantInfo(RoundParticipantEntity participant) {
        this.roundParticipantId = participant.getId();
        this.userId = participant.getUser().getUserId();
        this.picture = participant.getUser().getPicture();
        this.statusTag = participant.getStatusTag();
        this.taskProgress = participant.calculateTaskProgress();
    }

    public static RoundParticipantInfo from(RoundParticipantEntity participant){
        return new RoundParticipantInfo(participant);
    }
}
